ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -314,4 +306,22 @@ - (NSDictionary *)constantsToExport
     resolve(@(batteryLevel));
 }
 
+RCT_EXPORT_METHOD(getUserAgent:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (_webView == nil) {
+            _webView = [[WKWebView alloc] initWithFrame:CGRectZero];
+        }
+        [_webView evaluateJavaScript:@"navigator.userAgent" completionHandler:^(id result, NSError * error) {
+            if (error == nil) {
+                if (result != nil) {
+                    resolve([NSString stringWithFormat:@"%@", result]);
```

Comment:
[docs](https://developer.apple.com/documentation/webkit/wkwebview/1415017-evaluatejavascript) say:

> [...] The completion handler always runs on the main thread.

perhaps it's better to resolve this not on main thread?
can we utilize `self.methodQueue`?

Reply:
I didn't create a `methodQueue` only for that so I dispatch on global priority default queue like shown in https://facebook.github.io/react-native/docs/native-modules-ios.html#threading